packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createOrUpdateIndividualConstituent/index.ts

```diff
@@ -0,0 +1,778 @@
+              '@path': '$.traits.website'
+            },
+            then: {
+              '@path': '$.traits.website'
+            },
+            else: {
+              '@path': '$.properties.website'
+            }
+          }
+        },
+        primary: '',
+        type: ''
+      }
+    },
+    phone: {
+      label: 'Phone',
+      description: "The constituent's phone number.",
+      type: 'object',
+      properties: {
+        do_not_call: {
+          label: 'Do Not Call',
+          type: 'boolean'
+        },
+        number: {
+          label: 'Phone Number',
+          type: 'string'
+        },
+        primary: {
+          label: 'Is Primary',
+          type: 'boolean'
+        },
+        type: {
+          label: 'Phone Type',
+          type: 'string'
+        }
+      },
+      default: {
+        do_not_call: '',
+        number: {
+          '@if': {
+            exists: {
+              '@path': '$.traits.phone'
+            },
+            then: {
+              '@path': '$.traits.phone'
+            },
+            else: {
+              '@path': '$.properties.phone'
+            }
+          }
+        },
+        primary: '',
+        type: ''
+      }
+    }
+  },
+  perform: async (request, { payload }) => {
+    // search for existing constituent
+    let constituentId = undefined
+    if (payload.email?.address || payload.lookup_id) {
+      // default to searching by email
+      let searchField = 'email_address'
+      let searchText = payload.email?.address
+
+      if (payload.lookup_id) {
+        // search by lookup_id if one is provided
+        searchField = 'lookup_id'
+        searchText = payload.lookup_id
+      }
+
+      try {
+        const constituentSearchResponse = await request(
+          `${SKY_API_BASE_URL}/constituents/search?search_field=${searchField}&search_text=${searchText}`,
+          {
+            method: 'get'
+          }
+        )
+        const constituentSearchResults = await constituentSearchResponse.json()
+
+        if (constituentSearchResults.count > 1) {
+          // multiple existing constituents, throw an error
+          throw new IntegrationError('Multiple records returned for given traits', 'MULTIPLE_EXISTING_RECORDS', 400)
+        } else if (constituentSearchResults.count === 1) {
+          // existing constituent
+          constituentId = constituentSearchResults.value[0].id
+        } else if (constituentSearchResults.count !== 0) {
+          // if constituent count is not >= 0, something went wrong
+          throw new IntegrationError('Unexpected record count for given traits', 'UNEXPECTED_RECORD_COUNT', 500)
+        }
+      } catch (error) {
+        const statusCode = error?.response?.status
+        const errorMessage = statusCode
+          ? `${statusCode} error occurred when searching for constituent`
+          : 'Error occurred when searching for constituent'
+        if (statusCode === 401 || statusCode === 429 || statusCode >= 500) {
```

Comment:
should 401 be retryable?

Reply:
I added 401 with the thought that if something happens that causes the auth token to start failing, the customer can go into settings and fix it, then retry anything that failed.

Reply:
I think the only scenario I'm afraid of is if the application-level credentials fail and cause a huge backlog of requests. Let's confirm with Joe what sort of loads the retrying system can handle.